.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,51 @@
+name: build_commits
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: build commit code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@37335c7bb261b353407cff977110895fa0b4f7d8
+        with:
+          go-version: 1.16.3
+          
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@v2.1.4
+        with:
+          path: $HOME/.clang
+          key: llvm-10.0
+          
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1.2.1
+        with:
+          version: "10.0"
+          directory: $HOME/.clang
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+          
+      - name: Install libelf
+        run: |
+          sudo apt-get install libelf-dev
+          
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          
+      - name: Check if build works for every commit
+        run: |
+          PR_COMMITS_API_JSON=$(curl \
+            -H "Accept: application/vnd.github.v3+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            ${{ github.event.pull_request.commits_url }})
+          PR_FIRST_SHA=$(echo "$PR_COMMITS_API_JSON" | jq -r ".[0].sha")
+          PR_PARENT_SHA=$(git rev-parse "${PR_FIRST_SHA}^")
+          git rebase --exec "make clean && make -j $(nproc) && make -C bpf build_all -j $(nproc)" $PR_PARENT_SHA
```

Comment:
Looks like the action is failing because of email and name not being set in git config? Setting them to dummy values should be good enough for this workflow.

Reply:
Yes this should be the issue as the commit was running fine in the forked. Thanks for pointing.

Reply:
I was thinking of using the name and email of the Pull requester instead of dummy. Like using 
`git config --global user.name "$(git --no-pager log --format=format:'%an' -n 1)"` and `git config --global user.email "$(git --no-pager log --format=format:'%ae' -n 1)"`
Would it be better?

Reply:
FWIW, this would resolve to the last commit author on the PR rather than the PR author. If you do want the PR author, you should probably retrieve information from the PR event payload (I think there is stuff in `${{ github.event.pull_request.user }}`) or the PR API again.

But to be honest since these values won't be used for real (it's a rebase that won't actually be pushed), I feel like this is overengineering. I would probably just setup Git config in a separate step at the top with bogus values such as:
```sh
git config user.name "GitHub Actions"
git config user.email "github-actions@users.noreply.github.com"
```